### PR TITLE
add query functionality to the /api/reviews endpoint using TDD accoun…

### PR DESCRIPTION
**Pull request by JakeMaciver**
…ting for errors

### CODE REVIEW

File: __tests__/endpoint.test.js [hunk 1]
```diff
@@ -138,6 +138,54 @@ describe('GET /api/reviews', () => {
...
+  });
+  test('404: should return a 404 error if user tries to query a column that does exist', () => {
+    return request(app).get('/api/reviews?category=edsgh').expect(404)
+  });
+  test('404: should return a 404 error if user enters an order that is not whitelisted', () => {
+    return request(app).get('/api/reviews?order=fshg').expect(404)
+  });
+  test('404: should return a 404 error if user enters a sort_by that is not whitelisted', () => {
+		return request(app).get('/api/reviews?sort_by=fshg').expect(404);
+	});
```
Comment (johnhutcheon) [inline __tests__/endpoint.test.js:185]: These 2 tests (order and sort by) should be 400 instead of 404

File: __tests__/endpoint.test.js [hunk 2]
```diff
@@ -138,6 +138,54 @@ describe('GET /api/reviews', () => {
...
+  test('404: should return a 404 error when the category exists but there are no reviews associated with it', () => {
+    return request(app).get("/api/reviews?category=children's games").expect(404)
+  });
 });
```
Comment (johnhutcheon) [inline __tests__/endpoint.test.js:195]: For this test, rather than a 404, we should receive a 200 with an empty array of reviews.

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Jake. Nice work so far, but could you just see my comments regarding the tests please and resubmit? Thank you! 🙂

File: __tests__/endpoint.test.js [hunk 3]
```diff
@@ -138,6 +138,60 @@ describe('GET /api/reviews', () => {
...
+			.expect(200)
+			.then(({ body }) => {
+				const { reviews } = body;
+				expect(reviews).toBeSortedBy('created_at', { descending: true });
+			});
+  });
+  test('404: should return a 404 error if user tries to query a column that does exist', () => {
+    return request(app).get('/api/reviews?category=edsgh').expect(404)
+  });
+  test('400: should return a 404 error if user enters an order that is not whitelisted', () => {
```
Comment (pippaamy) [inline __tests__/endpoint.test.js:180]: remember to change the description to 400 here too

File: __tests__/endpoint.test.js [hunk 4]
```diff
@@ -138,6 +138,60 @@ describe('GET /api/reviews', () => {
...
+		return request(app).get('/api/reviews?order=fshg').expect(400);
+	});
+  test('400: should return a 404 error if user enters a sort_by that is not whitelisted', () => {
+		return request(app).get('/api/reviews?sort_by=fshg').expect(400);
```
Comment (pippaamy) [inline __tests__/endpoint.test.js:184]: and here

**Review by pippaamy — APPROVED** [review]
Good work, just change those descriptions and you're all good to merge!

**Pull request merged**